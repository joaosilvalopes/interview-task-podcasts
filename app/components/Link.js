import { Link } from '@remix-run/react';
import styled from 'styled-components';

const _Link = styled(Link)`
    text-decoration: none;
    color: inherit;

    &:focus, &:hover, &:visited, &:link, &:active {
        text-decoration: none;
    }
`;

export default _Link;
