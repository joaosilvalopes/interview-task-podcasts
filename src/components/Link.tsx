import { Link } from "react-router-dom";
import styled from 'styled-components';

const _Link = styled(Link)`
    text-decoration: none;
    color: inherit;

    &:focus, &:hover, &:visited, &:link, &:active {
        text-decoration: none;
    }
`;

export default _Link;
