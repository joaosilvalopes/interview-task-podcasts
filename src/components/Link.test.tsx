import { render } from '@testing-library/react';
import { RouterProvider, createBrowserRouter } from "react-router-dom";

import Link from './Link';

it('should render correctly', async () => {
    const props = { to: "/" };
    const router = createBrowserRouter([{ path: "/", element: <Link { ...props } /> }]);
    const { asFragment, container } = render(<RouterProvider router={router} />);
    
    expect(container.firstElementChild?.getAttribute?.('href')).toBe('/');
    expect(asFragment()).toMatchSnapshot();
});
