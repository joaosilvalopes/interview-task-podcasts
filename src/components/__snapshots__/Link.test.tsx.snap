// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <a
    class="sc-bcXHqe iklKes"
    href="/"
  />
</DocumentFragment>
`;
